content/300-guides/050-database/900-turso.mdx

```diff
@@ -0,0 +1,183 @@
+- **Making schema changes**. Since libSQL uses HTTP to connect to the remote database, this makes it incompatible with Prisma Migrate. However, you can use [`prisma migrate diff`](/reference/api-reference/command-reference#migrate-diff) <span class="api"></span> to create a schema migration and then apply the changes to your database using [Turso's CLI](https://docs.turso.tech/reference/turso-cli).
+
+## Connect and query your primary database
+
+1. Create a database on Turso if you don't have one already:
+
+   ```sh
+   turso db create turso-prisma-db
+   ```
+
+2. Retrieve the database's connection string:
+
+   ```sh
+   turso db show turso-prisma-db
+   ```
+
+3. Create an authentication token that will allow you to connect to the database:
+
+   ```sh
+   turso db tokens create turso-prisma-db
+   ```
+
+4. Update your `.env` file with the authentication token and connection string:
+
+   ```text
+   TURSO_AUTH_TOKEN="eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9..."
+   TURSO_DATABASE_URL="libsql://turso-prisma-db-user.turso.io"
+   ```
+
+5. Enable the `driverAdapters` Preview feature flag:
+
+   ```prisma highlight=3;add
+   generator client {
+     provider        = "prisma-client-js"
+     previewFeatures = ["driverAdapters"]
+   }
+   
+   datasource db {
+     provider = "sqlite"
+     url      = "file:./dev.db"
+   }
+   ```
+
+6. Generate Prisma Client:
+
+   ```sh
+   npx prisma generate
+   ```
+
+7. Install the libSQL database client and Prisma driver adapter for libSQL packages:
+
+   ```sh
+   npm install @libsql/client @prisma/adapter-libsql
+   ```
+
+8. Update your Prisma Client instance with the following snippet:
+
+   ```ts
+   import { PrismaClient } from '@prisma/client'
+   import { PrismaLibSQL } from '@prisma/adapter-libsql'
+   import { createClient } from '@libsql/client'
+
+   const libsql = createClient({
+     url: `${process.env.TURSO_DATABASE_URL}`,
+     authToken: `${process.env.TURSO_AUTH_TOKEN}`,
+   })
+
+   const adapter = new PrismaLibSQL(libsql)
+   const prisma = new PrismaClient({ adapter })
+   ```
+
+You can use Prisma Client as you normally would with full type-safety in your project.
+
+## How to manage schema changes
+
+Prisma Migrate and Introspection workflows are currently not supported when working with Turso. This is because Turso uses HTTP to connect to your database, which Prisma Migrate doesn't support.
+
+To update your database schema:
+
+1. Generate a migration file using `prisma migrate dev` against a local SQLite database:
+
+```sh
+npx prisma migrate dev --name init # Migration name
+```
+
+2. Apply the migration using Turso's CLI:
+
+```sh
+turso db shell turso-prisma-db < ./prisma/migrations/20230922132717_init/migration.sql # Replace `20230922132717_init` with the existing migration
+```
```

Comment:
Surprise, the formatting is broken here:
<img width="160" alt="image" src="https://github.com/prisma/docs/assets/183673/f9647bc4-d75f-4fd5-9a10-889df5f22e9c">